code/uilib/uipulldownmenucontainer.cpp

```diff
@@ -143,9 +143,16 @@ UIPulldownMenuContainer::~UIPulldownMenuContainer()
 {
 	for (int i = m_popups.NumObjects(); i > 0; i--)
 	{
-		uipopup_describe* uid = m_popups.ObjectAt(i);
+		uipopup_describe* uipd = m_popups.ObjectAt(i);
 		m_popups.RemoveObjectAt(i);
-		delete uid;
+		if (uipd->data)
```

Comment:
A better place for this code would be inside `uipopup_describe`'s destructor (create one if it doesn't exist)